aana/api/request_handler.py

```diff
@@ -229,3 +230,46 @@ async def _async_chat_completions(
                 "created": int(time.time()),
                 "choices": [{"index": 0, "message": response["message"]}],
             }
+
+    @app.get("/api/status", response_model=SDKStatusResponse)
+    async def status(self) -> SDKStatusResponse:
+        """The endpoint for checking the status of the application.
+
+        Returns:
+            SDKStatusResponse: The response containing the status of the application.
+        """
+        serve_status = serve.status()
+        message = ""
+        if any(
+            app.status == "DEPLOY_FAILED"
+            or app.status == "UNHEALTHY"
+            or app.status == "NOT_STARTED"
+            for app in serve_status.applications.values()
+        ):
+            sdk_status = SDKStatus.UNHEALTHY
+            error_messages = []
+            for app_name, app_status in serve_status.applications.items():
+                if (
+                    app_status.status == "DEPLOY_FAILED"
+                    or app_status.status == "UNHEALTHY"
+                ):
+                    for (
+                        deployment_name,
+                        deployment_status,
```

Comment:
How would be the status of a deployment with more that 1 replica? Will it be "UNHEALTHY" if one of replica failed?

Reply:
Yes, if one of the replicas is unhealthy, the whole app is unhealthy. This is default Ray behaviour.